auto-complete.js

```diff
@@ -117,50 +134,67 @@ var autoComplete = (function(){
             }
             that.parentElement.appendChild(that.sc);
 
-            live('autocomplete-suggestion', 'mouseleave', function(e){
+            live('autocomplete-suggestion', 'mouseleave', function (e) {
                 var sel = that.sc.querySelector('.autocomplete-suggestion.selected');
-                if (sel) setTimeout(function(){ sel.className = sel.className.replace('selected', ''); }, 20);
+                if (sel) setTimeout(function () { sel.className = sel.className.replace('selected', ''); }, 20);
             }, that.sc);
 
-            live('autocomplete-suggestion', 'mouseover', function(e){
+            live('autocomplete-suggestion', 'mouseover', function (e) {
                 var sel = that.sc.querySelector('.autocomplete-suggestion.selected');
                 if (sel) sel.className = sel.className.replace('selected', '');
                 this.className += ' selected';
             }, that.sc);
 
-            live('autocomplete-suggestion', 'mousedown', function(e){
-                if (hasClass(this, 'autocomplete-suggestion')) { // else outside click
-                    var v = this.getAttribute('data-val');
+            live('autocomplete-suggestion-text', 'mousedown', function (e) {
+                if (hasClass(this, 'autocomplete-suggestion-text')) { // else outside click
+                    var v = this.parentElement.getAttribute('data-val');
+                    var index = this.parentElement.getAttribute('data-index');
```

Comment:
`parentElement`? Why the HTML structure changed here? This will break for lib users that didn't change the structure to this new one.

Reply:
Really, that structure can break.
I made a change that can fix this